src/Test.UnitTests.Sarif/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -33,21 +33,30 @@ protected override string ConstructTestOutputFromInputResource(string inputResou
 
             SarifLog actualLog = PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(transformedLog, formatting: Formatting.None, out transformedLog);
 
-            Uri originalUri = actualLog.Runs[0].OriginalUriBaseIds["TESTROOT"].Uri;
-            string uriString = originalUri.ToString();
+            // For CoreTests only - this code rewrites the log persisted URI to match the test environment
+            if (inputResourceName == "Inputs.CoreTests.sarif")
+            {
+                Uri originalUri = actualLog.Runs[0].OriginalUriBaseIds["TESTROOT"].Uri;
+                string uriString = originalUri.ToString();
+
+                string currentDirectory = Environment.CurrentDirectory;
+                currentDirectory = currentDirectory.Substring(0, currentDirectory.IndexOf(@"\bld\"));
+                uriString = uriString.Replace("REPLACED_AT_TEST_RUNTIME", currentDirectory);
 
-            // This code rewrites the log persisted URI to match the test environment
-            string currentDirectory = Environment.CurrentDirectory;
-            currentDirectory = currentDirectory.Substring(0, currentDirectory.IndexOf(@"\bld\"));
-            uriString = uriString.Replace("REPLACED_AT_TEST_RUNTIME", currentDirectory);
+                actualLog.Runs[0].OriginalUriBaseIds["TESTROOT"] = new ArtifactLocation { Uri = new Uri(uriString, UriKind.Absolute) };
 
-            actualLog.Runs[0].OriginalUriBaseIds["TESTROOT"] = new ArtifactLocation { Uri = new Uri(uriString, UriKind.Absolute) };
+                var visitor = new InsertOptionalDataVisitor(_currentOptionallyEmittedData);
+                visitor.Visit(actualLog.Runs[0]);
 
-            var visitor = new InsertOptionalDataVisitor(_currentOptionallyEmittedData);
-            visitor.Visit(actualLog.Runs[0]);
+                // Restore the remanufactured URI so that file diffing matches
+                actualLog.Runs[0].OriginalUriBaseIds["TESTROOT"] = new ArtifactLocation { Uri = originalUri };
+            }
+            else
+            {
+                var visitor = new InsertOptionalDataVisitor(_currentOptionallyEmittedData);
```

Comment:
>var visitor = new InsertOptionalDataVisitor(_currentOptionallyEmittedData); [](start = 16, length = 75)

the new test case doesnt actually have access to the original file, so i simply created an else case - i know this code could have been refactored better, i was rabit-holing a bit, so pushing this one in the interest of time. If you have suggestions on improvements, i want to hear :)